spec/support/integration.rb

```diff
@@ -97,6 +97,17 @@ def wait_until_topics_exist(broker, topics)
     end
   end
 
+  def run_in_thread(*args, &block)
```

Comment:
I misunderstood what it was that ran in the thread when I saw this in the tests. I thought it was the block. Maybe call it something that conveys "run this command in the background". You can also use `#value` on thread instead of `#join` to make sure that exceptions that kill the thread are visible in the tests (it also makes the thread object look less like a thread and you can call it "future" to make things look less low level, if you want to).